rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -146,13 +139,13 @@ public void setNewFederation(Federation federation) {
 
     @Override
     public Federation getOldFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
-        if (oldFederation != null || shouldSaveOldFederation) {
-            return oldFederation;
+        if (oldFederationTracker.isPresent() || oldFederationTracker.isModified()) {
```

Comment:
Isn't the second condition here useless? What would be the case where `oldFederationTracker` is empty but `isModified`?  It would return null, is it what we expect?

Reply:
It is, not clear the background but it is the expected behavior. If the method 'setFederation' is called passing null as a federation, it should keep the flag "isModified" as it was